src/theory/arith/nl/transcendental/sine_solver.cpp

```diff
@@ -266,13 +266,27 @@ void SineSolver::checkMonotonic()
                                d_data->d_zero,
                                d_data->d_pi_neg_2,
                                d_data->d_pi_neg};
-  std::vector<Node> mpoints_vals;
+  // sound upper, lower bounds for the above points
+  std::vector<Node> mpointsBound[2];
 
   // get model values for points
-  for (const auto& point : mpoints)
+  TNode tpi = d_data->d_pi;
```

Comment:
Can you add a comment on what exactly `mpointsBound` contains afterwards?
I think `mpointsBound[0/1][x]` is an upper/lower bound on `mpoints[x]`, but I'm really not sure.
Below there is a comment that indicates it is actually lower/upper...